### PR TITLE
Refactoring the Supercluster rendering for easy testing

### PR DESCRIPTION
**Pull request by goemen**
The supercluster render in the PointCluster component was easy to test, interacting directly with leaflet in the react tests is not any playground. I refactored the component to use react component in the render. 

I used this refactored code to write unit tests, showcasing how we can write more tests in the leaflet component

### CODE REVIEW

File: frontend/src/components/maps/leaflet/InventoryLayer.tsx
Comment (Fosol) [inline frontend/src/components/maps/leaflet/InventoryLayer.tsx:75]: Removes listener?

File: frontend/src/components/maps/leaflet/InventoryLayer.tsx
Comment (Fosol) [inline frontend/src/components/maps/leaflet/InventoryLayer.tsx:118]: Is this because the linter doesn't understanding `useDeepCompareEffect(...)`?

File: frontend/src/components/maps/leaflet/mapUtils.tsx
Comment (Fosol) [inline frontend/src/components/maps/leaflet/mapUtils.tsx:103]: Documentation

File: frontend/src/components/maps/leaflet/Spiderfier.ts
Comment (Fosol) [inline frontend/src/components/maps/leaflet/Spiderfier.ts:69]: I don't understand this change?

File: frontend/src/components/maps/leaflet/PointClusterer.tsx
Comment (Fosol) [inline frontend/src/components/maps/leaflet/PointClusterer.tsx:113]: Why was this change necessary?

**Review by Fosol — APPROVED** [review]

File: frontend/src/components/maps/leaflet/MapProperties.test.tsx
Comment (agalbrai) [inline frontend/src/components/maps/leaflet/MapProperties.test.tsx:165]: what's the difference between wait() and await act()?

**Review by agalbrai — COMMENTED** [review]
I'm a little fuzzy on all the details so I won't approve it but I assume it works well :)

File: frontend/src/components/maps/leaflet/Map.tsx
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/Map.tsx:226]: Can you confirm that this doesn't adversely affect the 'zoom' functionality please?

File: frontend/src/components/maps/leaflet/MapProperties.test.tsx
Comment (devinleighsmith) [reply frontend/src/components/maps/leaflet/MapProperties.test.tsx:165]: wait is used to allow an expectation to finish before continuing. Behind the scenes the wait runs in a loop, and just ignores any failures or errors until the timeout is reached. 

act is used to wrap an action that will cause a change in a rendered component. For example, clicking a submit button. Sometimes the act will also need to check for something to be true before exiting.

Generally use wait if you are waiting for something that isn't a direct result of one of your actions. Use act otherwise.

File: frontend/src/components/maps/leaflet/MapProperties.test.tsx
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/MapProperties.test.tsx:178]: Ah, so this ended up being required?

File: frontend/src/components/maps/leaflet/MapProperties.test.tsx
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/MapProperties.test.tsx:219]: How is this test working with the mock being removed? does it throw a warning?

File: frontend/src/components/maps/leaflet/PointClusterer.tsx
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/PointClusterer.tsx:117]: what is a crime point?

File: frontend/src/components/maps/leaflet/PointClusterer.tsx
Comment (devinleighsmith) [reply frontend/src/components/maps/leaflet/PointClusterer.tsx:113]: This is a pretty big render block with a lot of different categories of functionality. Any way we could split this up? Even just creating functions for these for readability would be fine.

File: frontend/src/features/properties/map/MapView.tsx
Comment (devinleighsmith) [inline frontend/src/features/properties/map/MapView.tsx:94]: Hmm, what's going on here.

File: frontend/src/hooks/useApi.ts
Comment (devinleighsmith) [inline frontend/src/hooks/useApi.ts:120]: Please at least some basic error handling logic since this is a core application request that fails periodically.

Comment (devinleighsmith) [reply frontend/src/hooks/useApi.ts:120]: Also I think this will not display the loading bar anymore right? Refactoring is great but lets not regress functionality.

**Comment by devinleighsmith** [comment]
@goemen I think this needs to be demoed to @Fosol or myself as this PR contains a lot of map changes that aren't covered by existing tests.

**Review by devinleighsmith — CHANGES_REQUESTED** [review]
Please setup a demo for this before merging.

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**
